test-cypress/cypress/commands/api/visits.ts

```diff
@@ -5,39 +5,117 @@ import { v1ApiPath, getCreds } from "../server";
 import { logTestDescription } from "../descriptions";
 
 Cypress.Commands.add(
-  "checkVisits",
-  (user: string, camera: string, noVists: number) => {
-    logTestDescription(`Check number of visits is ${noVists}`, {
-      user: user,
-      camera: camera,
-      expectedVisits: noVists
+  "checkVisitTags",
+  (user: string, camera: string, expectedTags: string[]) => {
+    const expectedVisits = expectedTags.map((tag) => {
+      return { tag };
     });
 
-    const where: VisitsWhere = {
-      duration: { $gte: "0" },
-      type: "thermalRaw"
-    };
+    logTestDescription(
+      `Check visit tags match ${JSON.stringify(expectedTags)}`,
+      {
+        user,
+        camera,
+        expectedVisits
+      }
+    );
 
-    if (camera) {
-      where.DeviceId = getCreds(camera).id;
-    }
+    checkVisitsMatch(user, camera, expectedVisits);
+  }
+);
 
-    const params = {
-      where: JSON.stringify(where),
-      limit: 100
-    };
-
-    const visitsNum = noVists;
-    cy.request({
-      method: "GET",
-      url: v1ApiPath("recordings/visits", params),
-      headers: getCreds(user).headers
-    }).then((response) => {
-      expect(response.body.numVisits).to.eq(visitsNum);
+Cypress.Commands.add(
+  "checkVisits",
+  (user: string, camera: string, expectedVisits: ComparableVisit[]) => {
+    logTestDescription(`Check visits match ${JSON.stringify(expectedVisits)}`, {
+      user,
+      camera,
+      expectedVisits
     });
+
+    checkVisitsMatch(user, camera, expectedVisits);
   }
 );
 
+function checkVisitsMatch(
+  user: string,
+  camera: string,
+  expectedVisits: ComparableVisit[]
+) {
+  const where: VisitsWhere = {
+    duration: { $gte: "0" },
+    type: "thermalRaw"
+  };
+
+  if (camera) {
+    where.DeviceId = getCreds(camera).id;
+  }
+
+  const params = {
+    where: JSON.stringify(where),
+    limit: 100
+  };
+
+  cy.request({
+    method: "GET",
+    url: v1ApiPath("recordings/visits", params),
+    headers: getCreds(user).headers
+  }).then((response) => {
+    checkResponseMatches(response, expectedVisits);
+  });
+}
+
+function checkResponseMatches(
+  response: any,
+  expectedVisits: ComparableVisit[]
+) {
+  const responseVisits = response.body.visits;
+
+  expect(
+    responseVisits.length,
+    `Number of visits to be ${responseVisits.length}`
+  ).to.eq(expectedVisits.length);
+  const increasingDateResponseVisits = responseVisits.reverse();
```

Comment:
Just remember that `.reverse()` on an array in JS also mutates the original array in place - which may or may not be what you expect here.